### ISSUE TITLE
Dialogflow with corporate proxy

### ISSUE THREAD

**Issue by anhem115** [googleapis/nodejs-dialogflow#698]
Hi, I'm using the latest version of dialogflow, with the basic example to detect an intent. I could make it run normally without proxy setting but got problem with proxy.
Here is the version in my package.json
```
"dependencies": {
    "@google-cloud/dialogflow": "^3.3.0",
    "@grpc/grpc-js": "1.1.7",
    "@types/node": "^14.11.2",
    "body-parser": "^1.19.0",
    "cors": "^2.8.5",
    "express": "^4.17.1",
    "save": "^2.4.0",
    "uuid": "^8.3.1"
  },
```
I use the following environment variable:
http_proxy: myproxyserver.com:80
https_proxy: myproxyserver.com:80
grpc_proxy: myproxyserver.com:80

Here is my GRPC log:
```
2020-10-21T16:48:46.025Z | proxy | Proxy server myproxyserver.com set by environment variable grpc_proxy
2020-10-21T16:48:46.027Z | proxy | No proxy server list set by environment variable no_proxy
2020-10-21T16:48:46.029Z | resolving_load_balancer | dns:myproxyserver.com IDLE -> IDLE
2020-10-21T16:48:46.030Z | connectivity_state | dns:myproxyserver.com IDLE -> IDLE
2020-10-21T16:48:46.032Z | dns_resolver | Resolver constructed for target dns:myproxyserver.com
2020-10-21T16:48:46.045Z | channel | dns:myproxyserver.com createCall [0] method="/google.cloud.dialogflow.v2.Sessions/DetectIntent", deadline=Wed Oct 21 2020 11:49:46 GMT-0500 (Central Daylight Time)
2020-10-21T16:48:46.050Z | call_stream | [0] Sending metadata
2020-10-21T16:48:46.052Z | channel | Pick result: QUEUE subchannel: undefined status: undefined undefined
2020-10-21T16:48:46.066Z | call_stream | [0] write() called with message of length 102
2020-10-21T16:48:46.067Z | call_stream | [0] end() called
2020-10-21T16:48:46.070Z | call_stream | [0] deferring writing data chunk of length 107
2020-10-21T16:48:46.071Z | dns_resolver | Resolution update requested for target dns:myproxyserver.com
2020-10-21T16:48:46.075Z | resolving_load_balancer | dns:myproxyserver.com IDLE -> CONNECTING
2020-10-21T16:48:46.078Z | channel | Pick result: QUEUE subchannel: undefined status: undefined undefined
2020-10-21T16:48:46.078Z | connectivity_state | dns:myproxyserver.com IDLE -> CONNECTING
2020-10-21T16:48:46.078Z | resolving_load_balancer | dns:myproxyserver.com CONNECTING -> CONNECTING
2020-10-21T16:48:46.079Z | channel | Pick result: QUEUE subchannel: undefined status: undefined undefined
2020-10-21T16:48:46.079Z | connectivity_state | dns:myproxyserver.com CONNECTING -> CONNECTING
2020-10-21T16:48:46.083Z | dns_resolver | Resolved addresses for target dns:myproxyserver.com: [xxx.xxx.xxx.xxx:443]
2020-10-21T16:48:46.094Z | pick_first | Connect to address list xxx.xxx.xxx.xxx:443
2020-10-21T16:48:46.097Z | subchannel | xxx.xxx.xxx.xxx:443 refcount 0 -> 1
2020-10-21T16:48:46.097Z | subchannel | xxx.xxx.xxx.xxx:443 refcount 1 -> 2
2020-10-21T16:48:46.098Z | pick_first | Start connecting to subchannel with address xxx.xxx.xxx.xxx:443
2020-10-21T16:48:46.099Z | pick_first | IDLE -> CONNECTING
2020-10-21T16:48:46.100Z | resolving_load_balancer | dns:myproxyserver.com CONNECTING -> CONNECTING
2020-10-21T16:48:46.101Z | channel | Pick result: QUEUE subchannel: undefined status: undefined undefined
2020-10-21T16:48:46.101Z | connectivity_state | dns:myproxyserver.com CONNECTING -> CONNECTING
2020-10-21T16:48:46.102Z | subchannel | xxx.xxx.xxx.xxx:443 IDLE -> CONNECTING
2020-10-21T16:48:46.104Z | proxy | Using proxy xxx.xxx.xxx.xxx:443 to connect to dialogflow.googleapis.com:443
2020-10-21T16:48:46.114Z | pick_first | CONNECTING -> CONNECTING
2020-10-21T16:48:46.115Z | resolving_load_balancer | dns:myproxyserver.com CONNECTING -> CONNECTING
2020-10-21T16:48:46.115Z | channel | Pick result: QUEUE subchannel: undefined status: undefined undefined
2020-10-21T16:48:46.115Z | connectivity_state | dns:myproxyserver.com CONNECTING -> CONNECTING
Failed to connect to proxy xxx.xxx.xxx.xxx:443 with error connect ECONNREFUSED xxx.xxx.xxx.xxx:443
2020-10-21T16:48:46.126Z | subchannel | xxx.xxx.xxx.xxx:443 CONNECTING -> TRANSIENT_FAILURE
2020-10-21T16:48:46.127Z | pick_first | CONNECTING -> TRANSIENT_FAILURE
2020-10-21T16:48:46.127Z | resolving_load_balancer | dns:myproxyserver.com CONNECTING -> TRANSIENT_FAILURE
2020-10-21T16:48:46.128Z | channel | Pick result: TRANSIENT_FAILURE subchannel: undefined status: 14 No connection established
2020-10-21T16:48:46.128Z | call_stream | [0] cancelWithStatus code: 14 details: "No connection established"
2020-10-21T16:48:46.129Z | call_stream | [0] ended with status: code=14 details="No connection established"
2020-10-21T16:48:46.130Z | connectivity_state | dns:myproxyserver.com CONNECTING -> TRANSIENT_FAILURE
2020-10-21T16:48:46.217Z | channel | dns:myproxyserver.com createCall [1] method="/google.cloud.dialogflow.v2.Sessions/DetectIntent", deadline=Wed Oct 21 2020 11:49:46 GMT-0500 (Central Daylight Time)
```
I could see it tried to connect through myproxyserver.COM:443 instead of myproxyserver.COM:80. I'm not sure whether we could have a way to fix that. Could you help me on that one @bcoe 

Updated:
Because my proxyserver is up at port 80 (http://myproxyserver.com:80), then it will be removed when the proxy url is parsed at `http_proxy.js`  
```
proxyUrl = new url_1.URL(proxyEnv);
```

Then it will be processed through the `resolver-dns.js` which will attach a default port 443 if there is no port in the proxy server.
```
this.ipResult = [
                    {
                        host: hostPort.host,
                        port: (_a = hostPort.port) !== null && _a !== void 0 ? _a : DEFAULT_PORT,
                    },
                ];
```
So in this situation, I have either: set up my proxy server to serve at a different port or modify the default port in the `resolver-dns.js`.

Do you have any idea about that ?
_Originally posted by @anhem115 in https://github.com/googleapis/nodejs-dialogflow/issues/20#issuecomment-713719235_

**Comment by anhem115** [comment]
Closed since this issue is more relevant to grpc-js repository

**Comment by anhem115** [comment]
Update: This issue is fixed for the next version of grpc-js (1.1.8 apparently)
